Tvheadend: support zipped content from web

mhutzsch opened the pull request:
unzip content from web download if the download is zipped

some epg data is available as zip, e.g. https://rytec.ricx.nl/epg_data/rytecDE_Basic.gz with this fix the content is unzipped when it is downloaded directly.

chewitt commented:
Please increment the add-on version in `package.mk` and place a comment in the changelog file, else when the change is merged it will be built and pushed to the repo but Kodi will not see a newer add-on version to update to.

chewitt commented:
Final task is to squash the commits into a single commit and reword the commit message (not the PR title) to follow house style; e.g. `tvheadend42: and support for zipped epg content`

mhutzsch commented:
Hi @chewitt apologies for missing the version bump and readme changes. had some troubles with my multiple git identities, but now finally I squashed all commits into one and updated the commit message. Thanks for the proposal how to change the commit message. Really awesome how quick you react on a PR!

chewitt commented:
You happen to have caught us (me) on a morning where the emails pinged at the right intervals between other tasks. I'm sure we can be normally-slow in future PRs :)

CvH commented on packages/addons/service/tvheadend42/source/bin/tv_grab_file:
```diff
@@ -37,7 +37,14 @@ then
   elif [ "$XMLTV_TYPE" = "WEB" ]; then
-    wget -qO - "$XMLTV_LOCATION_WEB"
+    case "$XMLTV_LOCATION_WEB" in
+    *.gz | *.bz2 | *.xz)
```
did you tested it with xz and bz2 ? wondering if gunzip is actually able to do that at all

mhutzsch replied:
you are right. I did this in the morning before I started my real-life work. I hope I am not such a bad developer, when I am at work :|
piping the wget output into zcat should fix this and handle all types of data - equally to the handler for compressed files. This bash script is not DRY know - to avoid other sideeffects I did not change any other parts of the script and it is still short and easy readable. Sorry for annoying you with that minor change, @chewitt

CvH replied:
we use busybox zcat, that looks like it supports bz2 and xz
`wget -qO - https://tukaani.org/xz/xz-5.2.5.tar.xz | zcat`
`wget -qO - https://tukaani.org/xz/xz-5.2.5.tar.bz2 | zcat`

looks working :)

CvH replied:
@mhutzsch it looks working (I have nothing for testing in real world), its okay to merge ?

mhutzsch replied:
Yeah, merging should have no negative side effect.

CvH reviewed (approved):
tx
